src/jobs/jobs.controller.ts

```
@@ -297,150 +311,131 @@ export class JobsController {
-          pid: { $in: datasetIds },
-          ownerGroup: { $in: user.currentGroups },
-        },
-      });
-      const datasetsNotOwner = datasetIds.length - numberOfDatasets.count;
+    jobConfiguration: any
+  ): Promise<JobClass> {
 
-      if (datasetsNotOwner > 0) {
-        throw new HttpException(
-          {
-            status: HttpStatus.BAD_REQUEST,
-            message:
-              "Unauthorized access to " +
-              datasetsNotOwner +
-              " datasets out of " +
-              datasetIds.length,
-          },
-          HttpStatus.BAD_REQUEST,
-        );
+
+    // NOTE: We need JobClass instance because casl module works only on that. 
+    // If other fields are needed can be added later.
+    const jobInstance = new JobClass();
+    jobInstance._id = "";
+    jobInstance.ownerUser = "";
+    jobInstance.ownerGroup = "";
+    jobInstance.accessGroups = [];
+    jobInstance.requesterEmail = jobCreateDto.requesterEmail;
+    jobInstance.datasetsValidation = false;
+    jobInstance.configuration = jobConfiguration;
+    jobInstance.statusCode = "Initializing";
+    jobInstance.statusMessage = "Building and validating job, verifying authorization";
+
+    if (user) {
+      // the request comes from a user who is logged in.
+      if (user.currentGroups.some((g) => configuration().adminGroups.includes(g))) {
+        // admin users
+        let jobUser: JWTUser | null = user;
+        if (user.username != jobCreateDto.ownerUser) {
+          jobUser = await this.usersService.findByUsername2JWTUser(jobCreateDto.ownerUser);
+        }
+
+        jobInstance.ownerUser = jobUser?.username as string;
+        jobInstance.requesterEmail = jobUser?.email as string;
+        jobInstance.ownerGroup = (
+          jobCreateDto.ownerGroup
+            ? jobCreateDto.ownerGroup
+            : jobUser?.currentGroups[0]
+        )
+      } else {
+        jobInstance.ownerUser = user.username;
+        jobInstance.requesterEmail = user.email;
```

Should throw an error if it differs